cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp

```diff
@@ -112,10 +112,17 @@
                 institutionLoginUrl = "${okstateUrl}";
             } else if (institutionLoginUrl === "cord") {
                 institutionLoginUrl = "${cordUrl}"
+            } else {
+                // Removing the fragment part of the login URL is not necessary but a good-to-have. Thus, neither will
+                // we run into nor do we need to worry about weird corner cases if somehow it breaks login endpoints.
+                let lastIndexOfHash = institutionLoginUrl.lastIndexOf("#");
+                if (lastIndexOfHash >= 0) {
+                    institutionLoginUrl = institutionLoginUrl.substring(0, lastIndexOfHash);
+                }
```

Comment:
The fragment is removed at the front-end when users click the login button. This is redundant but good-to-have. Another note is that this only applies to `saml-shib` institutions since `cas-pac4j` ones don't have the login URL.